cmd/frontend/internal/auth/userpasswd/handlers.go

```diff
@@ -418,15 +418,26 @@ func HandleCheckUsernameTaken(logger log.Logger, db database.DB) http.HandlerFun
 			return
 		}
 		if err != nil {
-			httpLogAndError(logger, w, "Error checking username uniqueness", http.StatusInternalServerError, log.Error(err))
+			httpLogError(logger, log.LevelError, w, "Error checking username uniqueness", http.StatusInternalServerError, log.Error(err))
 			return
 		}
 
 		w.WriteHeader(http.StatusOK)
 	}
 }
 
-func httpLogAndError(logger log.Logger, w http.ResponseWriter, msg string, code int, errArgs ...log.Field) {
-	logger.Error(msg, errArgs...)
+func httpLogError(logger log.Logger, level log.Level, w http.ResponseWriter, msg string, code int, errArgs ...log.Field) {
```

Comment:
An idea: 

```suggestion
func httpLogError(logFunc func(string, ...log.Field), w http.ResponseWriter, msg string, code int, errArgs ...log.Field) {
```

then:

```go
httpLogError(logger.Warn, w, err, http.StatusUnauthorized)
```

(also, the log entry should include the code!)

Reply:
Damn, of course that's much simpler haha. Thanks!